### ISSUE TITLE
Add Brainpool256R1

### ISSUE THREAD

**Issue by drj-e** [latchset/jwcrypto#207]
Many thanks for this great library.

I would need to use 'Brainpool256R1' for signing a JWT and to encrypt to JWE.
Is there a guide for adding this algorithm in all files?

**Comment by simo5** [comment]
I am not aware of any RFC describing Brainpool algorithms for JWT, what standard would you be following?

**Comment by drj-e** [comment]
It is a requirement of the telematics of the German healthcare system, no RFC standard.

**Comment by simo5** [comment]
At the very least a draft should be published otherwise there is no way to agree on interoperable naming.
Names are transported inside the JWT, so without an official name you can't interoperate and your JWTs will be considered faulty by any implementation that does not use your naming convention.

See IANA Considerations [Section 7 of RFC 7518](https://datatracker.ietf.org/doc/html/rfc7518#section-7).

For JWcrypto I can accept something "unofficial", those curves will not be available in the default set. Users will have to explicitly enable them, but there is precedent to do it.

The other thing is that names may change in future should an official RFC come out for those curves with different names so I would suggest using a name like EBP256_u (U for unofficial) for the algorithm_registry in jwa.py until a name is registered to IANA.

For the rest just follow the same conventions used for something like ES256 in jwa.py

for jwk I would use UnofficialBrainpool256R1 in JWKEllipticCurveRegistry
and add into JWKpycaCurveMap {'brainpoolp256r1': 'brainpoolp256r1'} ,
as well as add in _get_curve_by_name() the righ ec.BrainpoolP256R1() invocation.
Basically grep for 'secp256k1' and do the same stuff just with brainpoolp256r1 anywhere you find that string or it's mappings (SECP256K1, 'SECG secp256k1 curve', ..)

The main issue I see in the end is the naming, because that wil go in tokens as the Curve Name, and it will be for non-standard non-interoperable tokens, and will be painful if later on a different name is standardized upon.

**Comment by drj-e** [comment]
Thank you very much. I will implement the advice and test whether it can meet the recipient's requirements.
